harvesters/src/main/java/org/fao/geonet/kernel/harvest/harvester/database/DatabaseHarvesterAligner.java

```diff
@@ -0,0 +1,408 @@
+import org.fao.geonet.utils.Xml;
+import org.jdom.Element;
+import org.springframework.util.StringUtils;
+
+import java.util.*;
+import java.util.concurrent.atomic.AtomicBoolean;
+
+import static org.fao.geonet.kernel.harvest.harvester.csw.Aligner.applyBatchEdits;
+
+class DatabaseHarvesterAligner extends BaseAligner<DatabaseHarvesterParams> implements IHarvester<HarvestResult> {
+    private Logger log;
+    private ServiceContext context;
+    private DataManager dataMan;
+    private IMetadataManager metadataManager;
+    private IMetadataUtils metadataUtils;
+    private IMetadataIndexer metadataIndexer;
+    private IMetadataSchemaUtils metadataSchemaUtils;
+    private HarvestResult result;
+    private CategoryMapper localCateg;
+    private GroupMapper localGroups;
+    private UUIDMapper localUuids;
+    private List<HarvestError> errors = new LinkedList<>();
+    private List<Integer> idsForHarvestingResult;
+    private String processName;
+    private Map<String, Object> processParams = new HashMap<>();
+
+    @Override
+    public List<HarvestError> getErrors() {
+        return errors;
+    }
+
+    public DatabaseHarvesterAligner(AtomicBoolean cancelMonitor, Logger log, ServiceContext context, DatabaseHarvesterParams params) {
+        super(cancelMonitor);
+        this.log = log;
+        this.context = context;
+        this.params = params;
+
+        result = new HarvestResult();
+        result.addedMetadata = 0;
+        result.uuidSkipped = 0;
+        result.datasetUuidExist = 0;
+        result.couldNotInsert = 0;
+
+        GeonetContext gc = (GeonetContext) context.getHandlerContext(Geonet.CONTEXT_NAME);
+        dataMan = gc.getBean(DataManager.class);
+        metadataManager = gc.getBean(IMetadataManager.class);
+        metadataSchemaUtils = gc.getBean(IMetadataSchemaUtils.class);
+        metadataUtils = gc.getBean(IMetadataUtils.class);
+        metadataIndexer = gc.getBean(IMetadataIndexer.class);
+    }
+
+    @Override
+    public HarvestResult harvest(Logger log) throws Exception {
+        this.log = log;
+        if (log.isDebugEnabled()) {
+            log.debug("Retrieving remote metadata information for : " + params.getName());
+        }
+
+        DatabaseMetadataRetriever metadataRetriever = DatabaseMetadataRetrieverFactory.getDatabaseMetadataRetriever(params.getDatabaseType(), params.getServer(), params.getPort(),
+            params.getDatabase(), params.getUsername(), params.getPassword(), log);
+
+        log.info("Start of alignment for : " + params.getName());
+        result = new HarvestResult();
+        //----------------------------------------------------------------
+        //--- retrieve all local categories and groups
+        //--- retrieve harvested uuids for given harvesting node
+        localCateg = new CategoryMapper(context);
+        localGroups = new GroupMapper(context);
+        localUuids = new UUIDMapper(context.getBean(IMetadataUtils.class), params.getUuid());
```

Comment:
Is it possible to use the one defined line 92 ?

Reply:
Done in https://github.com/geonetwork/core-geonetwork/pull/8247/commits/0d74819fa131f8f61455bc462420a305864d7c0c